Use "nested flow" in place of "subflow" in the docs (#14122)
PrefectHQ/prefect: docs/2.19.x/concepts/flows.mdx

When the nested flow run completes, the task runner is shut down.

Nested flow runs block execution of their parent flow run until completion. 
However, asynchronous nested flow runs can be execute concurrently by using [AnyIO task groups](https://anyio.readthedocs.io/en/stable/tasks.html) or [asyncio.gather](https://docs.python.org/3/library/asyncio-task.html#id6).

Nested flows differ from normal flows in that they will resolve any passed task futures into data. This allows data to be passed from the parent flow to the child easily.

The relationship between a child and parent flow is tracked by creating a special task run in the parent flow. This task run will mirror the state of the child flow run.

A task that represents a nested flow will be annotated as such in its `state_details` via the presence of a `child_flow_run_id` field. A nested flow can be identified via the presence of a `parent_task_run_id` on `state_details`.

You can define multiple flows within the same file. Whether running locally or via a [deployment](https://docs.prefect.io/concepts/deployments/), you must indicate which flow is the entrypoint for a flow run.
<Warning>
**Cancelling nested flow runs**

> Inline nested flow runs, specifically those created without `run_deployment`, cannot be cancelled without cancelling their parent flow run. If you may need to cancel a nested flow run independent of its parent flow run, we recommend deploying it separately and starting it using the [run\_deployment](https://docs.prefect.io/api-ref/prefect/deployments/deployments/#prefect.deployments.deployments.run_deployment) function.

Reviewer: ```suggestion
Inline nested flow runs cannot be cancelled without cancelling their parent flow run. 
If you need to be able to cancel a nested flow run independent of its parent flow run, deploy it separately and start it using the [run\_deployment](https://docs.prefect.io/api-ref/prefect/deployments/deployments/#prefect.deployments.deployments.run_deployment) method.
```

Author: ```suggestion
An inline nested flow run cannot be cancelled without cancelling its parent flow run. 
If you need to be able to cancel a nested flow run independent of its parent flow run, deploy it separately and start it using the [run\_deployment](https://docs.prefect.io/api-ref/prefect/deployments/deployments/#prefect.deployments.deployments.run_deployment) function.
```

Author: Additional update @billpalombi

The final state of the flow is determined by its return value. The following rules apply:

*   If an exception is raised directly in the flow function, the flow run is marked as failed.
*   If the flow does not return a value (or returns `None`), its state is determined by the states of all of the tasks and nested flows within it.

> *   If _any_ task run or nested flow run failed, then the final flow run state is marked as `FAILED`.

Reviewer: ```suggestion
*   If _any_ child run failed, then the final flow run state is marked as `FAILED`.
```

Traceback (most recent call last):...
ValueError: I am bad task
20:57:51.787 | INFO    | Task run 'always_succeeds_task-c9014725-0' - Finished in state Completed()

> 20:57:51.808 | INFO    | Flow run 'impartial-gorilla' - Created nested flow run 'unbiased-firefly' for flow 'always-succeeds-flow'

Reviewer: ```suggestion
20:57:51.808 | INFO    | Flow run 'impartial-gorilla' - Created subflow run 'unbiased-firefly' for flow 'always-succeeds-flow'
```
